src/Enum.php

```diff
@@ -229,15 +214,18 @@ protected static function resolve(): array
         $duplicatedValues = array_filter(array_count_values(static::getValues()), function (int $count) {
             return $count > 1;
         });
-        $duplicatedIndices = array_filter(array_count_values(static::getIndices()), function (int $count) {
-            return $count > 1;
-        });
 
         if (! empty($duplicatedValues)) {
+            unset(self::$cache[$class]);
             throw new DuplicatedValueException(array_keys($duplicatedValues), static::class);
         }
 
+        $duplicatedIndices = array_filter(array_count_values(static::getIndices()), function (int $count) {
+            return $count > 1;
+        });
+
         if (! empty($duplicatedIndices)) {
+            unset(self::$cache[$class]);
```

Comment:
Can this line be removed? Or is it needed for the `Exception`?

Reply:
It's needed to let the two new tests in `tests/DuplicationEnumTest.php` pass. Otherwise the duplicated values/indices would stay in the cache and get returned the next time.

Reply:
Maybe we could have a static dedicated method on the `Enum` class that clears the cache explicitly instead of this sneaky line.

Reply:
Yeah, or I don't write into `static::$cache` but use a local variable until it's fully resolved.